app/services/concerns/xccdf/profile_rule_groups.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving rule references
+  module ProfileRuleGroups
+    extend ActiveSupport::Concern
+
+    included do
+      def save_profile_rule_groups
+        ::ProfileRuleGroup.transaction do
+          ::ProfileRuleGroup.import!(profile_rule_groups,
+                                     on_duplicate_key_update: {
+                                       conflict_target: %i[rule_group_id profile_id],
+                                       columns: %i[rule_group_id profile_id]
+                                     })
+
+          base = ::ProfileRuleGroup.joins(profile: :benchmark)
+                                   .where('profiles.parent_profile_id' => nil)
+
+          profile_rule_group_links_to_remove(base).delete_all
+        end
+      end
+
+      private
+
+      def profile_rule_groups
+        @profile_rule_groups ||= @op_profiles.flat_map do |op_profile|
+          profile_id = profile_id_for(ref_id: op_profile.id)
+          rule_group_ids_for(ref_ids: op_profile.selected_rule_ids).map do |rule_group_id|
```

Comment:
This `selected_rule_ids` is a bit misleading and I needed to manually verify if it also returns the groups. Ideally it should be separated into two methods as we're iterating through a superset of the required items, twice. I created [this ticket](https://issues.redhat.com/browse/RHICOMPL-2833) to track the resolution of this techdebt. No action is needed in this PR :wink: